go/log_messages.md

```diff
@@ -0,0 +1,90 @@
+## Log Messages
+
+### Resource Reconciliation
+
+When doing resource reconciliation action we often have to find out the current
+state, and apply changes to bring it towards desired state. This often means
+calling some API and then emitting debug logs about that. We should try to
```

Comment:
I think this is the most important part and not limited to reconciliation. We call APIs in other places as well and surrounding our calls with this kind of logging is not only helpful for operators IMO.